compiler/rustc_mir_transform/src/cost_checker.rs

```diff
@@ -69,31 +69,8 @@ impl<'tcx> Visitor<'tcx> for CostChecker<'_, 'tcx> {
                     NonDivergingIntrinsic::CopyNonOverlapping(..) => CALL_PENALTY,
                 };
             }
-            _ => self.super_statement(statement, location),
-        }
-    }
-
-    fn visit_rvalue(&mut self, rvalue: &Rvalue<'tcx>, _location: Location) {
-        match rvalue {
-            // FIXME: Should we do the same for `OverflowChecks`?
-            Rvalue::NullaryOp(NullOp::RuntimeChecks(RuntimeChecks::UbChecks), ..)
-                if !self
-                    .tcx
-                    .sess
-                    .opts
-                    .unstable_opts
-                    .inline_mir_preserve_debug
-                    .unwrap_or(self.tcx.sess.ub_checks()) =>
-            {
-                // If this is in optimized MIR it's because it's used later,
-                // so if we don't need UB checks this session, give a bonus
-                // here to offset the cost of the call later.
-                self.bonus += CALL_PENALTY;
```

Comment:
This bonus has disappeared now, right? Shouldn't we still add this as the later call is still present?
Cc @scottmcm